skia/RSkSVGContainerComponent.cpp

```diff
@@ -0,0 +1,145 @@
+ */
+
+#include <unordered_map>
+
+#include "include/core/SkPath.h"
+
+#include "ReactSkia/RSkThirdPartyFabricComponentsProvider.h"
+#include "RSkComponentRNSVGCircle.h"
+#include "RSkComponentRNSVGDefs.h"
+#include "RSkComponentRNSVGEllipse.h"
+#include "RSkComponentRNSVGGroup.h"
+#include "RSkComponentRNSVGLine.h"
+#include "RSkComponentRNSVGPath.h"
+#include "RSkComponentRNSVGRect.h"
+#include "RSkComponentRNSVGUse.h"
+#include "RSkComponentRNSVGSvgView.h"
+#include "RSkSVGContainerComponent.h"
+
+namespace facebook {
+namespace react {
+
+#define DOWN_CAST_TO_CLASS(object,class) dynamic_cast<class *>(object)
+
+RSkSVGContainerComponent::RSkSVGContainerComponent(const ShadowView &shadowView,
+                                                   RnsShell::LayerType layerType,
+                                                   SkSVGTag t)
+    : RSkComponent(shadowView,layerType),
+      RSkSVGNode(t) {}
+
+RSkSVGContainerComponent::~RSkSVGContainerComponent() {
+  childRSkNodeList_.clear();
+  nodeIDMapper_.reset();
+}
+
+RSkSVGNode* RSkSVGContainerComponent::getRSkSVGNodeForComponetWithName(std::shared_ptr<RSkComponent> component) {
+
+  static std::unordered_map<std::string, RSkSVGNode* (*)(std::shared_ptr<RSkComponent>)> s_mapStringValues={
+    { "RNSVGView",getRSKSVGNodeFor<RSkComponentRNSVGSvgView>},
+    { "RNSVGGroup",getRSKSVGNodeFor<RSkComponentRNSVGGroup>},
+    { "RNSVGCircle",getRSKSVGNodeFor<RSkComponentRNSVGCircle>},
+    { "RNSVGRect",getRSKSVGNodeFor<RSkComponentRNSVGRect>},
+    { "RNSVGEllipse",getRSKSVGNodeFor<RSkComponentRNSVGEllipse>},
+    { "RNSVGPath",getRSKSVGNodeFor<RSkComponentRNSVGPath>},
+    { "RNSVGLine",getRSKSVGNodeFor<RSkComponentRNSVGLine>},
+    { "RNSVGDefs",getRSKSVGNodeFor<RSkComponentRNSVGDefs>},
+    { "RNSVGUse",getRSKSVGNodeFor<RSkComponentRNSVGUse>}
+  };
+
+  std::string componentName=component->getComponentData().componentName;
+
+  if(s_mapStringValues.find(componentName.c_str()) != s_mapStringValues.end()) {
+    return s_mapStringValues[componentName.c_str()](component);
+  } else {
+    return nullptr;
+  }
+}
+
+void RSkSVGContainerComponent::mountChildComponent(
+    std::shared_ptr<RSkComponent> childComponent,
+    const int index) {
+  
+  RNS_LOG_INFO("Parent Componet :" << this->getComponentData().componentName<<" holding child :" << childComponent->getComponentData().componentName);
+
+  if(!childComponent || !childComponent.get()) {
+    RNS_LOG_ERROR("Invalid child component received");
+    return;
+  }
+
+  auto node=getRSkSVGNodeForComponetWithName(childComponent);
+  if(node) {
+    auto sksvgNode =sk_sp<SkSVGNode>(node);
+    RNS_LOG_INFO("Child Node type  : "<<(int)node->tag());
+    if(!node->nodeName.empty()) {
+    RNS_LOG_INFO("Child Node has ID : "<<node->nodeName);
+    nodeIDMapper_.set(SkString(node->nodeName),sksvgNode);
+    }
+    // If Child Node is of type of container , Merge it's defMap with it's Parent
+    if(DOWN_CAST_TO_CLASS(node,RSkSVGContainerComponent)) {
```

Comment:
optimization:
DOWN_CAST* is done twice , can we do it once and use the downcasted node